platform-services.md

```diff
@@ -0,0 +1,215 @@
++ [`semantics_mojom`](http://docs.flutter.io/flutter/semantics_mojom/semantics_mojom-library.html)
++ [`system_chrome_mojom`](http://docs.flutter.io/flutter/system_chrome_mojom/system_chrome_mojom-library.html)
++ [`system_sound_mojom`](http://docs.flutter.io/flutter/system_sound_mojom/system_sound_mojom-library.html)
+
+### Platform and Third-party Services for Flutter {#platform-and-third-party-services-for-flutter}
+
+To access APIs that are native to the iOS or Android mobile platform, Flutter provides the
+`FlutterView` interface. The `FlutterView` interface allows you to embed a bi-directional
+asynchronous messaging component in your platform code to send and receive native service calls
+in a transparent way. This flexible approach lets you create cross-platform mobile apps that can
+connect to platform-specific and third-party services on iOS and Android devices, while sharing a
+common user interface (UI) codebase in Dart.
+
+You can build your own integrations with platform-specific services and third-party SDKs. Ideally,
+the services that you invoke using the `FlutterView` interface should return non-graphical,
+data-only information. Examples include:
+
+*   Geolocation
+*   Contacts
+*   File system
+*   Network information
+*   Sensors
+
+You can also use the `FlutterView` interface to invoke APIs from third-party SDKs that run on iOS
+and Android devices.
+
+To use the `FlutterView` interface, you must add a platform-specific Flutter component into the
+platform’s view hierarchy. For example, on iOS, you add a `FlutterViewController` object to
+interact with a `UIView`.  Similarly, on Android, you add a `FlutterView` object to your app’s
+`android.view.View` hierarchy. Your app can then communicate between the Dart and iOS or
+Android-specific parts of the app by using a simple message-passing interface.
+
+### Message passing {#message-passing}
+
+Flutter offers a simple message-passing interface between your platform-specific code and the
+Dart code. The message passing interface is asynchronous, bi-directional, and supports optional
+reply routing. Messages consist of strings encoded in JSON format, and these is no facility for
+transferring handles to other objects along with messages. You are responsible for encoding and
+decoding messages sent through this mechanism.
+
+## Get the `hello_services` sample {#the-hello_services-sample}
+
+The Flutter SDK includes a reference sample project called
+[`hello_services`](https://github.com/flutter/flutter/tree/master/examples/hello_services)
+that demonstrates how to connect to the geolocation service within an iOS or Android application
+by using the `FlutterView` interface.
+
+The `hello_services` project comes with helper code to handle the passing of
+JSON-formatted messages to invoke platform services. If your app needs access to functionality
+provided by platform-specific iOS or Android services, we strongly encourage you to
```

Comment:
I don't think we "strongly encourage" that.  That's certainly one option, but we're likely to create a better onboarding experience in the future (e.g., via `flutter create`).  Instead, I would say that you "can" do that.